doc/xrcvd/transceiver-monitor-hld.md

```diff
@@ -61,27 +64,77 @@ New Transceiver info table and transceiver DOM sensor table will be added to sta
+    txbiaslowalarm          = FLOAT                              ; tx bias low alarm threshold
+    txbiashighwarning       = FLOAT                              ; tx bias high warning threshold
+    txbiaslowwarning        = FLOAT                              ; tx bias low warning threshold
+
+#### 1.1.3 Transceiver Status Table ####
+
+        ; Defines Transceiver Status info for a port
+        key                          = TRANSCEIVER_STATUS|ifname     ; Error information for SFP on port
+        ; field                      = value
+        status                       = 1*255VCHAR                    ; code of the SFP status (plug in, plug out)
+        error                        = 1*255VCHAR                    ; SFP error (N/A or a string like error1 | error2 )
+
+
+### 1.2 Accessing EEPROM from platform container ###
+
+Transceiver EEPROM information can be accessed via reading sysfs files(e.g. `/sys/bus/i2c/devices/2-0048/hwmon/hwmon4/qsfp9_eeprom`) or other ways, this is upon vendor's own implementation. 
 
+Transceiver EEPROM accessing can be achieved by legacy sfputil.py plugin or new platform API, Xcvrd supports both of these two methods. If platform API not yet implemented on some vendor's device, it will automatically fall back on sfputil.py plugin.
 
-### 1.2 Access eeprom from platform container ###
 
-Transceiver information eeprom can be accessed via read files(e.g. `/sys/bus/i2c/devices/2-0048/hwmon/hwmon4/qsfp9_eeprom`), different vendors may have these files under different folders, these folder need to be mounted to platform container so Xcvrd can access them. 
+### 1.3 Transceiver change event and vendor platform API###
 
+#### 1.3.1 Transceiver change event ####
 
-For the convenience of implementation and reduce the time consuming, need to do enhancement to the `SfpUtilBase` class:
+Currently 7 transceiver events are defined as below.
 
-1. `SfpUtilBase` internally should add the ability to read the eeprom and only pick up the interested bytes by given offset and number of bytes.
+    status='0' SFP removed,
+    status='1' SFP inserted,
+    status='2' I2C bus stuck,
+    status='3' Bad eeprom,
+    status='4' Unsupported cable,
+    status='5' High Temperature,
+    status='6' Bad cable.
 
-2. `SfpUtilBase` will provide APIs `get_eeprom_sfp_info_dict(self, port_num)` and `get_eeprom_dom_info_dict(self, port_num)` to return `eeprom_if_dict` and `eeprom_dom_dict` separately, the interested values of these two dict are defined  in section 1.1.1 and 1.1.2.  In these two APIs can pick up these values from eeprom by provide the corresponding offset and number of bytes. 
+However, multiple errors could exists at the same time. For example, a module can be unsupported cable and high temperature. The new transceiver event will be described in a bitmap. New transceiver definition below.
 
+    bit 32  : 0=SFP removed, 1=SFP inserted,

```

Comment:
>bit [](http://example.com/codeflow?start=4&length=3)

bit order is not defined? and index from 1?

Reply:
Yes, bit index starts from 1. And for bit order, 0x01 means bit 32 = 1.